docs/architecture/adr-043-nft-module.md

```diff
@@ -35,341 +36,224 @@ The current design is based on the work done by [IRISnet team](https://github.co
 
 We will create a module `x/nft`, which contains the following functionality:
 
-- Store and transfer NFTs utilizing `x/bank`.
-- Mint and burn NFTs.
-- Use `x/authz` to implement ERC721-style authorization.
+- Store NFTs and track their ownership.
+- Expose `Keeper` interface for composing modules to mint and burn NFTs.
+- Expose external `Message` interface for users to transfer ownership of their NFTs.
 - Query NFTs and their supply information.
 
 ### Types
 
-#### Metadata
+#### Genre
```

Comment:
Is there any reason not to use `Denom` instead of `Genre`? Not opposed to `Genre` beyond introducing new terminology when we already have terms that serve same or similar purpose.

Reply:
If anything, to keep this aligned with already established terminology to token ontologies, I would recommend using
`class` to define a set of tokens which share the same properties and behaviours. E.g. "cryptokitties" or "did:cosmos:kittynet:nft:cryptokitties"
`denom` COULD be a class identifier property. E.g. "CAT-0001" or "did:cosmos:kittynet:nft:cryptokitties:cat-0001"
`collection` COULD be used as a sub-class tag. E.g. "Blue-eyed" or did:cosmos:kittynet:nft:cryptokitties#blue-eyed"